x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,366 @@
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clicksave', {
+            defaultMessage: "Click 'Save' and confirm",
+          }),
+        ]
+      );
+    } else {
+      steps.push(
+        ...[
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.stopent', {
+            defaultMessage: 'Stop all your Enterprise Search nodes',
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.stopkibana', {
+            defaultMessage: 'Stop all your Kibana nodes',
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.removeconfig', {
+            defaultMessage: "Edit 'kibana.yml' to remove 'enterpriseSearch.host'",
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.restart', {
+            defaultMessage: 'Restart Kibana',
+          }),
+        ]
+      );
+    }
+    return [
+      {
+        level: 'critical',
+        deprecationType: 'feature',
+        title: i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.title', {
+          defaultMessage: 'Enterprise Search host(s) must be removed',
+        }),
+        message: i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.message', {
+          defaultMessage:
+            'Enterprise Search is not supported in versions >= 9.x. ' +
+            'You must remove Enterprise Search node(s) from your deployment to proceed with the upgrade. ' +
+            'Note that once Enterprise Search is stopped, products such as App Search, Workplace Search, and Elastic Crawler ' +
+            'will cease to operate. Native Connectors will also stop running syncs, between when Enterprise Search is stopped ' +
+            'and when the 9.x upgrade completes. For full details, see the documentation.',
+        }),
+        documentationUrl:
+          'https://elastic.co/guide/en/enterprise-search/current/upgrading-to-9-x.html',
+        correctiveActions: {
+          manualSteps: steps,
+        },
+      },
+    ];
+  } else {
+    return [];
+  }
+}
+
+/**
+ * if the customer was using Elastic Crawler, they must delete the connector records
+ */
+export async function getCrawlerDeprecations(
+  ctx: GetDeprecationsContext
+): Promise<DeprecationsDetails[]> {
+  const client = ctx.esClient.asInternalUser;
+  const crawlers: Connector[] = await fetchConnectors(client, undefined, true, undefined);
+  if (crawlers.length === 0) {
+    return []; // no deprecations to register if there are no Elastic Crawlers in the connectors index
+  } else {
+    return [
+      {
+        level: 'critical',
+        deprecationType: 'feature',
+        title: i18n.translate('xpack.enterpriseSearch.deprecations.crawler.title', {
+          defaultMessage:
+            'Elastic Crawler metadata records in the .elastic-connectors index must be removed.',
```

Comment:
```suggestion
            'Elastic Crawler metadata records in the `.elastic-connectors` index must be removed.',
```

Try to make sure consistency with backticks everywhere. Ideally, these would be formatted to actually render `code format`, but again probably fine tradeoff to save time/complexity here.